Tools/TestWebKitAPI/Tests/WebCore/ApplicationManifestParser.cpp

```diff
@@ -522,6 +527,28 @@ TEST_F(ApplicationManifestParserTest, Orientation)
     testOrientation("\"portrait-secondary\""_s, WebCore::ScreenOrientationLockType::PortraitSecondary);
 }
 
+TEST_F(ApplicationManifestParserTest, Lang)
+{
+    testLang("123"_s, String());
+    testLang("null"_s, String());
+    testLang("true"_s, String());
+    testLang("{ }"_s, String());
+    testLang("[ ]"_s, String());
+    testLang("\"\""_s, String());
+
+    // Invalid language tags should be ignored.
+    testLang("\"invalid-language-tag\""_s, String());
+
+    testLang("\"en\""_s, "en"_s);
+    testLang("\" en \""_s, "en"_s);
+    testLang("\"en-AU\""_s, "en-AU"_s);
+    testLang("\"zh-Hans-CN\""_s, "zh-Hans-CN"_s);
+    testLang("\"en-x-custom\""_s, "en-x-custom"_s);
+
+    // Language tags should be canonicalized.
+    testLang("\"DE-DE\""_s, "de-DE"_s);
```

Comment:
Maybe just add the following to canonicalize and trim: 
```suggestion
    testLang("\"  DE-DE  \""_s, "de-DE"_s);
```